main.go

```diff
@@ -109,29 +120,55 @@ func main() {
 		}
 	}()
 
-	if err := cmd.Wait(); err != nil {
+	err = cmd.Wait()
+	timeline = append(timeline, api.Timeline{
+		Event:     "test_end",
+		Timestamp: createTimestamp(),
+	})
+
+	if err != nil {
 		if exitError := new(exec.ExitError); errors.As(err, &exitError) {
 			exitCode := exitError.ExitCode()
 			if cfg.MaxRetries == 0 {
 				// If retry is disabled, we exit immediately with the same exit code from the test runner
+				sendMetadata(ctx, apiClient, cfg, timeline)
```

Comment:
This method returns an error, but since we don't store the return result or do anything with with it, then it just fails silently yeah?